contracts/contracts/solidity/StarknetValidator.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.6.0 <=0.8.0;
+
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorValidatorInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/TypeAndVersionInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AccessControllerInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol';
+import '@chainlink/contracts/src/v0.8/SimpleWriteAccessController.sol';
+
+import '@chainlink/contracts/src/v0.8/dev/vendor/openzeppelin-solidity/v4.3.1/contracts/utils/Address.sol';
+
+import '../../vendor/starkgate-contracts-solidity-v0.8/src/starkware/starknet/solidity/IStarknetMessaging.sol';
+import './utils/utils.sol';
+
+/**
+ * @title Validator - makes cross chain call to update the Sequencer Uptime Feed on L2
+ */
+contract Validator is TypeAndVersionInterface, AggregatorValidatorInterface, SimpleWriteAccessController {
+    int256 private constant ANSWER_SEQ_OFFLINE = 1;
+    /* Selector hardcoded because StarkNet generates selectors differently than the standard ethereum way
+    different hash function on stark curve is used */
+    uint256 constant STARK_SELECTOR_UPDATE_STATUS =
```

Comment:
how is this generated?

Reply:
I have a function get_selector.ts for that.

Reply:
is there a function that we can call on L1 to generate this instead of hardcoding it here?

Reply:
There is not, and it would be unnecessarily expensive because of the custom hash fn.